### ISSUE TITLE
error - data reading

### ISSUE THREAD

**Issue by khawar85** [spyder-ide/spyder#6837]
## Description

**What steps will reproduce the problem?**

1. 
2. 
3. 

**What is the expected output? What do you see instead?**

**Please provide any additional information below**

  File "/m/soft/anaconda/3/5.0.1/lib/python3.6/site-packages/spyder/widgets/variableexplorer/collectionseditor.py", line 449, in createEditor
    if not editor.setup_and_check(value, title=key, readonly=readonly):
  File "/m/soft/anaconda/3/5.0.1/lib/python3.6/site-packages/spyder/widgets/variableexplorer/arrayeditor.py", line 715, in setup_and_check
    self.current_dim_changed(self.last_dim)
  File "/m/soft/anaconda/3/5.0.1/lib/python3.6/site-packages/spyder/widgets/variableexplorer/arrayeditor.py", line 784, in current_dim_changed
    self.change_active_widget(0)
  File "/m/soft/anaconda/3/5.0.1/lib/python3.6/site-packages/spyder/widgets/variableexplorer/arrayeditor.py", line 764, in change_active_widget
    self.data[slice_index]))
IndexError: index 0 is out of bounds for axis 0 with size 0

## Version and main components

* Spyder Version: 3.2.4 
* Python Version: 3.6.3
* Qt Versions: 5.6.2, PyQt5 5.6 on Linux

## Dependencies
```
pyflakes >=0.6.0 :  1.6.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  0.20.3 (OK)
numpy >=1.7      :  1.13.3 (OK)
sphinx >=0.6.6   :  1.6.3 (OK)
rope >=0.9.4     :  0.10.5 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
psutil >=0.3     :  5.4.0 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
sympy >=0.7.3    :  1.1.1 (OK)
cython >=0.21    :  0.26.1 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
IPython >=4.0    :  6.1.0 (OK)
pylint >=0.25    :  1.7.4 (OK)

```

**Comment by ccordoba12** [comment]
Please describe step by step how you got this error so we can reproduce it in our side. If you fail to do that, we'll close this issue in a week from now.

**Comment by khawar85** [comment]
Dear Sir, 

I am trying to get back the original space dimensions from SOM .. self organizing maps. Still I have one extra dimension of neurons left. I am trying to get the last column of np.array. and when I want to see the outout with view variables. It has provided the error and said to automatically inform via GITHUB.

Thank you for help in advance

**Comment by CAM-Gerlach** [comment]
@khawar85 Thanks for following up. This is likely because the array is empty or otherwise was not initalized properly; if so, it is a duplicate of #5628 and was fixed in Spyder 3.2.5 which was released many months ago (current version is 3.2.8), so please update with ``conda update spyder`` from the command line. 

If that doesn't fix it. does it occur consistently on the same array? If so, can you save the array on which it occurs (e.g. with dump from pickle or joblib), close Spyder, and see if it occurs again if you reload it and do the same actions? If so, then if you're able to send us the saved array we can take it from here. Thanks!

**Comment by CAM-Gerlach** [comment]
No response; assumed duplicate of #5628 .
